typing/typedecl.ml

```diff
@@ -769,11 +784,11 @@ let check_well_founded_manifest env loc path decl =
   let args = List.map (fun _ -> Ctype.newvar()) decl.type_params in
   check_well_founded env loc path (Path.same path) (Ctype.newconstr path args)
 
-let check_well_founded_decl env loc path decl to_check =
+let check_well_founded_decl ?recmod env loc path decl to_check =
```

Comment:
Same here, it seems better to make ~recmod mandatory here (and on check_well_founded as well).